### DISCUSSION TITLE
How to use prompt hints?

### DISCUSSION THREAD

**Discussion by iwannabedev** [Blarc/ai-commits-intellij-plugin#155] [Q&A] [answered]
Hi,

Newbe here ;)

I've seen recent update with added feature: 
"Variable `{hint}' for prompt customization that is filled with content from commit message dialog."

...but I don't know how to use it. My understanding is that I type something I want to include in a prompt that's later sent to OpenAI API as {hint} variable in a prompt message. Is that correct? If so then it doesn't work, unless I'm missing something. I just type my hints into commit message field (I use PyCharm IDE), but it's being ignored and I don't see those hints in the final commit message generated by AI Commits.

**Comment by kidroca** [comment, marked as answer]
Hey, I've just tried it. There's a bit of a fine print at the bottom when you edit a prompt, saying 
> "Include a hint by using a dollar sign and prefix like this `{Here is a hint: $hint}`. This adds the hint text inside the curly brackets only if a hint is provided

So we have to use both curly braces and $, here's how I use it in my template:

```
...
{Author's notes: "$hint"}
...
{diff}
```

Then if I write something in the commit message box, like "code comments were outdated", it is injected as `Author's notes: "code comments were outdated"`

Some samples I've tried, so that I know it actually work - https://github.com/Blarc/ai-commits-intellij-plugin/discussions/18#discussioncomment-9150833 _(Though with that particular template I find my notes are ignored 50% of the time)_

**Reply by iwannabedev** [reply]
Thanks for that hint ;) I only used predefined prompts you provided and that's probably where the trouble comes from. I did more tests and indeed it works, but not always. I will make my own prompt or use one of those made by other users from that thread you mentioned. Thanks again!
